File: web/src/api/meta.ts
import { apiClient } from './client'

export interface MetaData {
  name: string
  version: string
  environment: string
  features: {
    dev_login: boolean
    scheduler: boolean
  }
}

interface MetaResponse {
  data: MetaData
  request_id: string
}

export async function fetchMeta(): Promise<MetaData> {
  const response = await apiClient.get<MetaResponse>('/meta')
  return response.data.data
}

File: sd-agent/sd_agent/api/meta.py
from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Request

from sd_agent import __version__

router = APIRouter(prefix="/api", tags=["meta"])


@router.get("/meta")
async def meta(request: Request) -> dict[str, Any]:
    settings = request.app.state.settings
    return {
        "data": {
            "name": settings.APP_NAME,
            "version": __version__,
            "environment": settings.ENV.value,
            "features": {
                "dev_login": settings.AUTH_DEV_LOGIN,
                "scheduler": settings.CRON_ENABLED,
                "outbox": settings.OUTBOX_ENABLED,
            },
        },
        "request_id": request.state.request_id,
    }
